packages/flutter/lib/src/material/combo_box.dart

```diff
@@ -0,0 +1,747 @@
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      setState(() {
+        leadingPadding = getWidth(_leadingKey);
+      });
+    });
+  }
+
+  double? getWidth(GlobalKey key) {
+    final BuildContext? context = key.currentContext;
+    if (context != null) {
+      final RenderBox box = context.findRenderObject()! as RenderBox;
+      return box.size.width;
+    }
+    return null;
+  }
+
+  List<ComboBoxEntry> filter(List<ComboBoxEntry> entries, TextEditingController textEditingController) {
+    final List<ComboBoxEntry> filteredEntries = entries.where((ComboBoxEntry entry)
+      => entry.label.toLowerCase().contains(textEditingController.text.toLowerCase())).toList();
+
+    return filteredEntries;
+  }
+
+  int? search(List<ComboBoxEntry> entries, TextEditingController textEditingController) {
+    for (int i = 0; i < entries.length; i++) {
+      final ComboBoxEntry entry = entries[i];
+      if (textEditingController.value.text.isNotEmpty
+          && entry.label.toLowerCase().contains(textEditingController.text.toLowerCase())) {
+        return i;
+      }
+    }
+
+    return null;
+  }
+
+  List<Widget> _buildButtons(
+    List<ComboBoxEntry> filteredEntries,
+    TextEditingController textEditingController,
+    TextDirection textDirection,
+    { int? focusedIndex }
+  ) {
+    final List<Widget> result = <Widget>[];
+    final double padding = leadingPadding ?? _kDefaultHorizontalPadding;
+    final ButtonStyle defaultStyle;
+    if (textDirection == TextDirection.ltr) {
+      defaultStyle = MenuItemButton.styleFrom(
+        padding: EdgeInsets.only(left: padding, right: _kDefaultHorizontalPadding),
+      );
+    } else {
+      defaultStyle = MenuItemButton.styleFrom(
+        padding: EdgeInsets.only(left: _kDefaultHorizontalPadding, right: padding),
+      );
+    }
+
+    for (int i = 0; i < filteredEntries.length; i++) {
+      final ComboBoxEntry entry = filteredEntries[i];
+      ButtonStyle effectiveStyle = entry.style ?? defaultStyle;
+      final Color foregroundColor = effectiveStyle.foregroundColor?.resolve(<MaterialState>{MaterialState.focused})
+        ?? Theme.of(context).colorScheme.onSurface;
+
+      // Simulate the focused state because the text field should always be focused
+      // during traversal.
+      effectiveStyle = entry.enabled && i == focusedIndex
+        ? effectiveStyle.copyWith(backgroundColor: MaterialStatePropertyAll<Color>(foregroundColor.withOpacity(0.12)))
+        : effectiveStyle;
+
+      final MenuItemButton menuItemButton = MenuItemButton(
+        style: effectiveStyle,
+        leadingIcon: entry.leadingIcon,
+        trailingIcon: entry.trailingIcon,
+        onPressed: entry.enabled ? () {
+          textEditingController.text = entry.label;
+          textEditingController.selection =
+              TextSelection.collapsed(offset: textEditingController.text.length);
+          currentHighlight = widget.enableSearch ? i : -1;
+        } : null,
```

Comment:
Might be easier to read:
```suggestion
        onPressed: entry.enabled
          ? () {
              textEditingController.text = entry.label;
              textEditingController.selection =
                  TextSelection.collapsed(offset: textEditingController.text.length);
              currentHighlight = widget.enableSearch ? i : -1;
            }
          : null,
```